tests/api2/test_audit_basic.py

```diff
@@ -0,0 +1,109 @@
+def initialize_for_smb_tests(request):
+    with dataset('audit-test-basic', data={'share_type': 'SMB'}) as ds:
+        with smb_share(os.path.join('/mnt', ds), 'AUDIT_BASIC_TEST', {
+            'purpose': 'NO_PRESET',
+            'guestok': False,
+            'audit': {'enable': True}
+        }) as s:
+            with user({
+                'username': SMBUSER,
+                'full_name': SMBUSER,
+                'group_create': True,
+                'password': PASSWD,
+                'smb': True
+            }) as u:
+                yield {'dataset': ds, 'share': s, 'user': u}
+
+
+def test_audit_config_defaults(request):
+    config = call('audit.config')
+    for key in [
+        'retention',
+        'quota',
+        'reservation',
+        'quota_fill_warning',
+        'quota_fill_critical',
+        'remote_logging_enabled',
+        'space'
+    ]:
+        assert key in config, str(config)
+
+    assert config['retention'] == 7
+    assert config['quota'] == 0
+    assert config['reservation'] == 0
+    assert config['quota_fill_warning'] == 80
+    assert config['quota_fill_critical'] == 95
+    assert config['remote_logging_enabled'] is False
+    for key in ['used', 'used_by_snapshots', 'used_by_dataset', 'used_by_reservation', 'available']:
+        assert key in config['space'], str(config['space'])
+
+    assert 'SMB' in config['enabled_services']
+
+
+def test_audit_config_updates(request):
+    """
+    This test just validates that setting values has expected results.
+    """
+    new_config = call('audit.update', {'retention': 10})
+    assert new_config['retention'] == 10
+
+    new_config = call('audit.update', {'quota': 1})
+    assert new_config['quota'] == 1
+
+    # Check that we're actually setting the quota by evaluating available space
+    # we should be within 1 Mib of quota target (sql database will already be written)
+    assert abs(new_config['space']['available'] - 1024 ** 3) < 1024 ** 2
+
+    new_config = call('audit.update', {'reservation': 1})
+    assert new_config['reservation'] == 1
+    assert new_config['space']['used_by_reservation'] != 0
+
+    new_config = call('audit.update', {
+        'quota_fill_warning': 70,
+        'quota_fill_critical': 80
+    })
+
+    assert new_config['quota_fill_warning'] == 70
+    assert new_config['quota_fill_critical'] == 80
+
+
+def test_audit_query(initialize_for_smb_tests):
+    share = initialize_for_smb_tests['share']
+    with smb_connection(
+        host=ip,
+        share=share['name'],
+        username=SMBUSER,
+        password=PASSWD,
+    ) as c:
+        fd = c.create_file('testfile.txt', 'w')
+        for i in range(0, 3):
+            c.write(fd, b'foo')
+            c.read(fd, 0, 3)
+        c.close(fd, True)
+
+    sleep(10)
+    ops = call('audit.query', {
+        'services': ['SMB'],
+        'query-filters': [['user', '=', SMBUSER]],
+        'query-options': {'count': True}
+    })
+    assert ops > 0
```

Comment:
Would it be a good idea to also create _another_ share which is **NOT** being audited, and then write to both shares, and verify that you only get events from the expected share?